<launch>
    <arg name="__ns_1" default="arm_1"/>
    <arg name="__ns_2" default="arm_2"/>
    <arg name="ip_1" default="192.168.1.196"/>
    <arg name="ip_2" default="192.168.1.220"/>
    <arg name="dual_arm" default="false"/>
    <param name="dual_arm" type="bool" value="$(arg dual_arm)" />

    <include file="$(find kitting_station)/launch/multimaster.launch" />

    <include file="$(find kitting_station)/launch/xarm.launch">
        <arg name="ip_1" value="$(arg ip_1)"/>
        <arg name="ip_2" value="$(arg ip_2)"/>
        <arg name="dual_arm" value="$(arg dual_arm)"/>
    </include>

    <group ns="$(arg __ns_1)">
        <param if="$(arg dual_arm)" name="link_name" type="string" value="L_link_base" />
        <param unless="$(arg dual_arm)" name="link_name" type="string" value="link_base" />
        <include file="$(find kitting_station)/launch/vision.launch" >
            <arg name="__ns" value="$(arg __ns_1)"/>
            <arg name="serial_no" value="819612071527" />
        </include>
        <include file="$(find kitting_station)/launch/transform.launch">
        <arg name="__ns" value="$(arg __ns_1)"/>
            <arg if="$(arg dual_arm)" name="parent_link" value="L_link_eef"/>
        </include>
    </group>

    <group if="$(arg dual_arm)" ns="$(arg __ns_2)">
        <param name="link_name" type="string" value="L_link_base" />
        <include file="$(find kitting_station)/launch/vision.launch" >
            <arg name="__ns" value="$(arg __ns_2)"/>
            <arg name="serial_no" value="934222070995" />
        </include>
        <include file="$(find kitting_station)/launch/transform.launch">
        <arg name="__ns" value="$(arg __ns_2)"/>
            <arg name="parent_link" value="R_link_eef"/>
        </include>
    </group>

    <include file="$(find kitting_station)/launch/programs.launch">
        <arg name="dual_arm" value="$(arg dual_arm)"/>
    </include>
</launch>